cGame_logic.cpp

```diff
@@ -547,38 +554,41 @@ void cGame::shakeScreenAndBlitBuffer() {
     if (TIMER_shake == 0) {
 		TIMER_shake = -1;
 	}
-	// blitSprite on screen
 
+	// blitSprite on screen
 	if (TIMER_shake > 0)
 	{
 		// the more we get to the 'end' the less we 'throttle'.
 		// Structure explosions are 6 time units per cell.
 		// Max is 9 cells (9*6=54)
 		// the max border is then 9. So, we do time / 6
-		if (TIMER_shake > 69) TIMER_shake = 69;
-
-		int offset = TIMER_shake / 5;
-		if (offset > 9)
-			offset = 9;
+    TIMER_shake = std::min(TIMER_shake, 69);
+    int offset = std::min(TIMER_shake / 5, 9);
 
 		shake_x = -abs(offset/2) + rnd(offset);
 		shake_y = -abs(offset/2) + rnd(offset);
 
-		blit(bmp_screen, bmp_throttle, 0, 0, 0+shake_x, 0+shake_y, screen_x, screen_y);
+		blit(bmp_screen, bmp_throttle, 0, 0, 0 + shake_x, 0 + shake_y, screen_x, screen_y);
+    vsync();
```

Comment:
if you remove vsync we get more fps. No need to worry about tearing due to double buffering technique.